iml-gui/crate/src/page/filesystem.rs

```diff
@@ -1,102 +1,183 @@
+                    },
+                );
             }
-            (mgt, mdt, ost)
-        },
-    );
 
-    mdt.sort_by(|a, b| natord::compare(&a.name, &b.name));
-    ost.sort_by(|a, b| natord::compare(&a.name, &b.name));
+            let (mgt, mut mdts, mut osts) = xs.into_iter().filter(|t| is_fs_target(model.id, t)).fold(
+                (vec![], vec![], vec![]),
+                |(mut mgt, mut mdts, mut osts), x| {
+                    match x.kind {
+                        TargetKind::Mgt => mgt.push(x),
+                        TargetKind::Mdt => mdts.push(x),
+                        TargetKind::Ost => osts.push(x),
+                    }
+                    (mgt, mdts, osts)
+                },
+            );
 
+            mdts.sort_by(|a, b| natord::compare(&a.name, &b.name));
+            osts.sort_by(|a, b| natord::compare(&a.name, &b.name));
+
+            model.mgt = mgt;
+            model.mdts = mdts;
+            model.osts = osts;
+        }
+        Msg::WindowClick => {
+            for (_, r) in model.rows.iter_mut() {
+                if r.dropdown.watching.should_update() {
+                    r.dropdown.watching.update();
+                }
+            }
+        }
+    }
+}
+
+pub(crate) fn view(cache: &ArcCache, model: &Model, all_locks: &Locks) -> Node<Msg> {
     div![
-        details_table(model, f),
-        T::wrapper_view(vec![
-            targets("Management Target", model, &mgt[..]),
-            targets("Metadata Targets", model, &mdt[..]),
-            targets("Object Storage Targets", model, &ost[..]),
-        ])
+        details_table(cache, all_locks, model),
+        targets("Management Target", cache, all_locks, &model.rows, &model.mgt[..]),
+        targets("Metadata Targets", cache, all_locks, &model.rows, &model.mdts[..]),
+        targets("Object Storage Targets", cache, all_locks, &model.rows, &model.osts[..]),
     ]
 }
 
-fn details_table<I>(model: &Model, f: &Filesystem) -> Node<I> {
+fn details_table(cache: &ArcCache, all_locks: &Locks, model: &Model) -> Node<Msg> {
+    let fs = cache.filesystem.get(&model.id).unwrap();
+
     div![
         class![C.bg_white, C.border_t, C.border_b, C.border, C.rounded_lg, C.shadow],
         div![
             class![C.flex, C.justify_between, C.px_6, C._mb_px, C.bg_gray_200],
             h3![
                 class![C.py_4, C.font_normal, C.text_lg],
-                format!("File system {}", f.label)
+                format!("Filesystem {}", &fs.label)
             ]
         ],
         T::wrapper_view(vec![
-            tr![T::th_right(plain!("Space Used / Total")), T::td_view(size_view(f))],
+            tr![T::th_left(plain!("Space Used / Total")), T::td_view(size_view(fs))],
             tr![
-                T::th_right(plain!("Files Created / Maximum")),
-                T::td_view(files_view(f))
+                T::th_left(plain!("Files Created / Maximum")),
```

Comment:
I aligned to the right, because I found it easier to read when the header and the value are closer to each other.

Reply:
I think we should be focusing on visual alignment between components rather than differing alignments dependant on type.

Reply:
That's true, but this table is in fact a vertical one in opposite to all others :)